src/_pytest/nodes.py

```diff
@@ -183,8 +193,10 @@ def listchain(self):
             starting from root of collection tree. """
         chain = []
         item = self
```

Comment:
I personally like the previous form better, since the loop condition is more visible. So I'd instead add `# type: Optional[Node]` here. But up to you.